### PR TITLE
Add utility method for transforming a ditto event message value into a generic structure

### PR DESCRIPTION
**Pull request by gboyvalenkov-bosch**
[#187] Add utility method for transforming a ditto event message value into a generic structure

- implemented

Signed-off-by: Georgi Boyvalenkov <Georgi.Boyvalenkov@bosch.io>

### CODE REVIEW

**Review by dimiter-todorov-georgiev — APPROVED** [review]

File: integration/util/web.go [hunk 1]
```diff
@@ -284,3 +284,12 @@ func GetLiveMessageTopic(fullThingID string, action protocol.TopicAction) string
 		WithAction(action)
 	return t.String()
 }
+
+// ConvertValue marshals an object(i.e map) and unmarshals the produced json into a specific structure
+func ConvertValue(value interface{}, result interface{}) error {
```
Comment (hristobojilov) [inline integration/util/web.go:289]: Why don't you use map type then?

Comment (hristobojilov) [inline integration/util/web.go:289]: Maybe Convert is better name here?

**Review by hristobojilov — CHANGES_REQUESTED** [review]

File: integration/util/web.go (hunk 1 above)
Comment (gboyvalenkov-bosch) [reply integration/util/web.go:289]: message values have interface{} type, and this way it is more generic.

Comment (gboyvalenkov-bosch) [reply integration/util/web.go:289]: @e-grigorov What do you think?

Comment (e-grigorov) [inline integration/util/web.go:289]: This method should be placed in its own file. It's too generic to be placed in web.go.

File: integration/util/web.go [hunk 2]
```diff
@@ -284,3 +284,12 @@ func GetLiveMessageTopic(fullThingID string, action protocol.TopicAction) string
 		WithAction(action)
 	return t.String()
 }
+
+// ConvertValue marshals an object(i.e map) and unmarshals the produced json into a specific structure
```
Comment (e-grigorov) [inline integration/util/web.go:288]: - `i.e.` - is not correct. It should be `e.g.` or removed
- json -> JSON

File: integration/util/web.go (hunk 1 above)
Comment (e-grigorov) [reply integration/util/web.go:289]: Sounds good to me. `Convert` is shorter and still meaningful.

File: integration/util/utils.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package util
+
+import "encoding/json"
+
+// Convert marshals an object(e.g map) and unmarshals the produced JSON into a specific structure
```
Comment (e-grigorov) [inline integration/util/utils.go:17]: - `e.g` -> `e.g.`
- even, the content can be clarified a little: marshals an object (e.g. map) to a JSON payload and unmarshals it to the given structure

File: integration/util/utils.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+func Convert(value interface{}, result interface{}) error {
```
Comment (e-grigorov) [inline integration/util/utils.go:18]: The parameters should be: from, to

File: integration/util/utils.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package util
```
Comment (e-grigorov) [inline integration/util/utils.go:13]: utils.go in util package is a little bit strange. What about common.go?

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: integration/util/web.go (hunk 1 above)
Comment (e-grigorov) [inline integration/util/web.go:289]: Now, this method can be removed.

**Review by hristobojilov — APPROVED** [review]

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**